api/notd/store/retriever.py

```diff
@@ -145,3 +146,26 @@ async def get_collection_by_address(self, address: str, connection: Optional[Dat
             raise NotFoundException(message=f'Collection with registry {address} not found')
         collection = collection_from_row(row)
         return collection
+
+    async def get_collection_holding(self, address: str, ownerAddress: str, connection: Optional[DatabaseConnection] = None) -> List[Token]:
+        boughtTokens = []
+        soldTokens= []
+        query = (
+            TokenTransfersTable.select()
+            .where(TokenTransfersTable.c.registryAddress == address)
+            .where(TokenTransfersTable.c.toAddress == ownerAddress)
+        )
+        query = query.where(TokenTransfersTable.c.toAddress == ownerAddress)
+        boughtResult = await self.database.execute(query=query, connection=connection)
+        boughtTokens = [(token.registryAddress, token.tokenId) for token in [token_transfer_from_row(row) for row in boughtResult]]
+        query = (
+            TokenTransfersTable.select()
+            .where(TokenTransfersTable.c.registryAddress == address)
+            .where(TokenTransfersTable.c.fromAddress == ownerAddress)
+        )
+        soldResult = await self.database.execute(query=query, connection=connection)
+        soldTokens = [(token.registryAddress, token.tokenId) for token in [token_transfer_from_row(row) for row in soldResult]]
+        uniqueBoughtTokens = set(boughtTokens)
+        uniqueSoldTokens = set(soldTokens)
+        tokensOwned = uniqueBoughtTokens - uniqueSoldTokens
```

Comment:
this doesnt make sense - if the user buys then sells then buys again this would be empty?